src/editor.js

```diff
@@ -1500,16 +1657,20 @@ export class OdooEditor extends EventTarget {
     }
 
     /**
-     * Prevent the pasting of HTML and paste text only instead.
+     * Handle safe pasting of html or plain text into the editor.
      */
     _onPaste(ev) {
         ev.preventDefault();
-        const pastedText = (ev.originalEvent || ev).clipboardData.getData('text/plain');
-        this.execCommand('insertText', pastedText);
+        const clipboardData = ev.clipboardData.getData('text/html');
+        if (clipboardData) {
+            this.execCommand('insertHTML', this._prepareClipboardData(clipboardData));
```

Comment:
If I drop a <p> inside a <p> it will indeed end up with <p><p>, which it shouldn't. This is a bug of `insertHTML` that should be addressed in another PR.